src/lib/components/DamageMeter.svelte

```diff
@@ -216,6 +217,25 @@
                 } else {
                     totalDamageDealt = encounter.encounterDamageStats.totalDamageDealt;
                 }
+                if ($settings.meter.timeUntilKill) {
+                    let remainingDps =
+                        players
+                            .filter(
+                                (e) => e.damageStats.damageDealt > 0 && !e.isDead && e.entityType != EntityType.ESTHER
+                            )
+                            .reduce((a, b) => a + b.damageStats.damageDealt, 0) /
+                        (duration / 1000);
+                    let remainingBossHealth = 0;
+                    if (encounter.currentBoss?.currentHp) {
+                        remainingBossHealth += encounter.currentBoss.currentHp;
+                    }
+                    if (encounter.currentBoss?.currentShield) {
+                        remainingBossHealth += encounter.currentBoss.currentShield;
+                    }
+                    let millisUntilKill = Math.floor((1000 * remainingBossHealth) / remainingDps);
```

Comment:
i don't understand why boss health is multiplied by 1000

Reply:
I could have also divided the remaining dps by 1000 to get it to dp(millis). it is just to make it work with the millistominutesandseconds function.